src/main/java/de/zalando/aruha/nakadi/controller/EventTypeController.java

```diff
@@ -0,0 +1,133 @@
+import de.zalando.aruha.nakadi.domain.EventType;
+import de.zalando.aruha.nakadi.problem.DuplicatedEventTypeNameProblem;
+import de.zalando.aruha.nakadi.problem.ValidationProblem;
+import de.zalando.aruha.nakadi.repository.DuplicatedEventTypeNameException;
+import de.zalando.aruha.nakadi.repository.EventTypeRepository;
+import de.zalando.aruha.nakadi.repository.NoSuchEventTypeException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.Errors;
+import org.springframework.web.bind.annotation.*;
+import org.zalando.problem.MoreStatus;
+import org.zalando.problem.Problem;
+
+import javax.validation.Valid;
+import javax.ws.rs.core.Response;
+
+import java.net.URI;
+import java.util.List;
+
+import static org.springframework.http.ResponseEntity.status;
+
+@RestController
+@RequestMapping(value = "/event_types")
+public class EventTypeController {
+
+    private EventTypeRepository repository;
+
+    @Autowired
+    public EventTypeController(EventTypeRepository repository) {
+        this.repository = repository;
+    }
+
+    @RequestMapping(method = RequestMethod.GET)
+    public ResponseEntity<?> list() {
+        List<EventType> eventTypes = repository.list();
+
+        return status(HttpStatus.OK).body(eventTypes);
+    }
+
+    @RequestMapping(method = RequestMethod.POST)
+    public ResponseEntity<?> createEventType(@Valid @RequestBody final EventType eventType, Errors errors) throws Exception {
+        if (errors.hasErrors()) {
+            Problem problem = new ValidationProblem(errors);
+
+            return status(HttpStatus.UNPROCESSABLE_ENTITY).body(problem);
+        } else {
+            return persist(eventType);
+        }
+    }
+
+    @RequestMapping(value = "/{name}", method = RequestMethod.PUT)
+    public ResponseEntity<?> update(
+            @PathVariable("name") final String name,
+            @RequestBody @Valid final EventType eventType,
+            Errors errors) {
+        if (errors.hasErrors()) {
+            Problem problem = new ValidationProblem(errors);
+
+            return status(HttpStatus.UNPROCESSABLE_ENTITY).body(problem);
+        }
+
+        try {
+            EventType existingEventType = repository.findByName(name);
+
+            validateName(name, eventType, errors);
+            validateSchema(eventType, existingEventType, errors);
+
+            if (!errors.hasErrors()) {
+                repository.update(eventType);
+
+                return status(HttpStatus.OK).build();
+            } else {
+                Problem problem = new ValidationProblem(errors);
+
+                return status(HttpStatus.UNPROCESSABLE_ENTITY).body(problem);
+            }
+        } catch (NoSuchEventTypeException e) {
+            Problem problem = Problem.
+                    builder().
+                    withType(URI.create("https://httpstatuses.com/404")).
```

Comment:
You don't need to do that, the Problem framework already sets the Type URI and title by the given StatusCode, afaik.
`Problem.valueOf(statusCode, detailMessage)` should be enough.